mne/io/brainvision/brainvision.py

```diff
@@ -334,7 +339,7 @@ def _read_vmrk(fname):
     return np.array(onset), np.array(duration), np.array(description), date_str
 
 
-def _event_id_func(desc, event_id, trig_shift_by_type):
+def _event_id_func(desc, event_id, trig_shift_by_type, warning_messages):
```

Comment:
Is returning through the parameters something that we do in the codebase?

I'm not sure we should add parameters to _event_id_func. Since it should be a function that takes str and returns int. see: 
https://github.com/mne-tools/mne-python/blob/50efb02eda5f9ea14e1e9ae4b6045b5e4bc0dba0/mne/annotations.py#L544-L550

but I gues it does the work. We can unify it afterwards.